is-valid-turn.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+const isValidMove = require("./is-valid-move");
+const makeMove = require("./make-move");
+const _ = require("lodash");
+
+function isValidTurn(gameState, diceRoll, proposedMoves) {
+    function getAceyDeucey() {
+        const aceyDeucey = {};
+        diceRoll.forEach(function(roll, index) {
+            if (roll === 1 && !aceyDeucey.hasOwnProperty("oneLocation")) {
+                aceyDeucey.oneLocation = index; 
+            } else if (roll === 2 && !aceyDeucey.hasOwnProperty("twoLocation")) {
+                aceyDeucey.twoLocation = index;
+            } else if (!aceyDeucey.hasOwnProperty("doublesLocation")) {
+                aceyDeucey.doublesLocation = index;
+            } else {
+                aceyDeucey.isAceyDeucey = false;
+            }
+        });
+        return aceyDeucey;
+    }
+    
+    function proposedMovesMatchDice() {
+        if (diceRoll.length > 2) {
+                      
+            const aceyDeucey = getAceyDeucey();
+            if (aceyDeucey.isAceyDeucey === false) {
+                return false;
+            }
+
+            if (
+                !(proposedMoves[0].numberOfSpaces === 1 && proposedMoves[1].numberOfSpaces === 2) &&
```

Comment:
What if you did something like:

```js
_(proposedMoves)
    .take(2) // grab the first two because that is all we care about
    .sortBy() // sort them so you do not have to care about the order when you do a check
    .isEqual([1, 2]) // do a deep equality check
```